java-checks/src/main/java/org/sonar/java/checks/MagicNumberCheck.java

```diff
@@ -57,6 +58,11 @@ public void scanFile(JavaFileScannerContext context) {
   }
 
   @Override
+  public void visitEnumConstant(EnumConstantTree tree) {
+    // Ignore literals within enum constant
```

Comment:
Check in the construction of the EnumConstantTreeImpl : https://github.com/SonarSource/sonar-java/blob/master/java-squid/src/main/java/org/sonar/java/model/declaration/EnumConstantTreeImpl.java#L32